sdk/core/ts-http-runtime/src/policies/auth/apiKeyAuthenticationPolicy.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { ApiKeyCredential } from "../../auth/credentials.js";
+import type { AuthScheme } from "../../auth/schemes.js";
+import type { PipelineRequest, PipelineResponse, SendRequest } from "../../interfaces.js";
+import type { PipelinePolicy } from "../../pipeline.js";
+import { ensureSecureConnection } from "./checkInsecureConnection.js";
+
+/**
+ * Name of the API Key Authentication Policy
+ */
+export const apiKeyAuthenticationPolicyName = "apiKeyAuthenticationPolicy";
+
+/**
+ * Options for configuring the API key authentication policy
+ */
+export interface ApiKeyAuthenticationPolicyOptions {
+  /**
+   * The credential used to authenticate requests
+   */
+  credential: ApiKeyCredential;
+  /**
+   * Optional authentication schemes to use. If not provided, schemes from the request will be used.
+   */
+  authSchemes?: AuthScheme[];
+  /**
+   * Allows for connecting to HTTP endpoints instead of enforcing HTTPS.
+   * CAUTION: Never use this option in production.
+   */
+  allowInsecureConnection?: boolean;
+}
+
+/**
+ * Gets a pipeline policy that adds API key authentication to requests
+ */
+export function apiKeyAuthenticationPolicy(
+  options: ApiKeyAuthenticationPolicyOptions,
+): PipelinePolicy {
+  return {
+    name: apiKeyAuthenticationPolicyName,
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      // Ensure allowInsecureConnection is explicitly set when sending request to non-https URLs
+      ensureSecureConnection(request, options);
+
+      const scheme = (request.authSchemes ?? options.authSchemes)?.find((x) => x.kind === "apiKey");
+
+      // Skip adding authentication header if no API key authentication scheme is found
+      if (!scheme) {
+        return next(request);
+      }
+      if (scheme.apiKeyLocation !== "header") {
```

Comment:
Who ends up setting this value? It's just interesting that we provide `header | query | cookie` as supported values but really only support header. Why do we need the other union members? Just curious

Reply:
This will be defined in the `TypeSpec` spec. The generator will generate from the spec and pass it through the policy. For this beta, we're only supporting `header` value, so this will fail at runtime for unsupported location. We have a tentative plan to support other locations in the future (I'll double check on this)